geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/SAddExecutor.java

```diff
@@ -30,49 +28,58 @@
 
 public class SAddExecutor extends SetExecutor {
 
+
   @Override
   public void executeCommand(Command command, ExecutionHandlerContext context) {
     List<byte[]> commandElems = command.getProcessedCommand();
 
-    long entriesAdded = 0L;
-
     if (commandElems.size() < 3) {
       command.setResponse(Coder.getErrorResponse(context.getByteBufAllocator(), ArityDef.SADD));
       return;
     }
 
     ByteArrayWrapper key = command.getKey();
-    try (AutoCloseableLock regionLock = withRegionLock(context, key)) {
-      Region<ByteArrayWrapper, Set<ByteArrayWrapper>> region = getRegion(context);
+//    Long entriesAdded = 0L;
+    Region<ByteArrayWrapper, Set<ByteArrayWrapper>> region = getRegion(context);
+
+    ByteArrayWrapper keyPlaceHolder = null;
 
-      Set<ByteArrayWrapper> entries = region.get(key);
-      if (entries == null) {
-        entries = new HashSet<>();
-      }
+    region.compute(
```

Comment:
FYI, compute on Region does not have a region specific implementation. It is implemented with the default impl from ConcurrentMap. So it basically does a region.get, executes your lambda, and does a region.put. It will retry but I think this will have the same concurrent modification issue you saw before because you modifying directly the "oldRegionData"

Reply:
It's been tweaked to modify a copy.